src/ecmascript/spidermonkey/document.c

```diff
@@ -492,3 +549,129 @@ document_writeln(JSContext *ctx, unsigned int argc, JS::Value *rval)
 {
 	return document_write_do(ctx, argc, rval, 1);
 }
+
+void
+string_replace(struct string *res, struct string *inp, struct string *what, struct string *repl)
+{
+	struct string tmp;
+	struct string tmp2;
+	char *head;
+	char *found;
+	char *ins;
+	char *tmp_cnt;
+	
+	init_string(&tmp);
+	init_string(&tmp2);
+
+	add_to_string(&tmp,inp->source);
+	
+	head = tmp.source;
+	int  count = 0;
+	ins = head;
+	if (what->length==0) 
+	{
+		add_to_string(res,inp->source); 
+		return; 
+	}
+
+	// count occurence of string in input
+	for (count = 0; tmp_cnt = strstr(ins, what->source); ++count) 
+	{
+		ins = tmp_cnt + what->length;
+	}
+	
+	for (int i=0;i<count;i++) {
+		// find occurence of string
+		found=strstr(head,what->source);
+		// count chars before and after occurence
+		int bf_len=found-tmp.source;
+		int af_len=tmp.length-bf_len-what->length;
+		// move head by what
+		found+=what->length;
+		// join the before, needle and after to res
+		add_bytes_to_string(&tmp2,tmp.source,bf_len);
+		add_bytes_to_string(&tmp2,repl->source,repl->length);
+		add_bytes_to_string(&tmp2,found,af_len);
+		// clear tmp string and tmp2 string
+		init_string(&tmp);
+		add_to_string(&tmp,tmp2.source);
```

Comment:
Here, before init_string(&tmp);
is required done_string(&tmp); and similar for tmp2.
init_string resets , but does not free memory of previous string.

I'm going to merge it tommorow.

Reply:
No problem. I'll fix that. Thank You. I'm looking forward to this to be commited.

Reply:
See below. When I free the string using done_string inside the loop elinks will dump the core.